specification/cognitiveservices/OpenAI.Inference/models/completions/extensions/oyd/vector_search.tsp

```diff
@@ -53,8 +53,10 @@ model OnYourDataEndpointVectorizationSource
   @doc("Specifies the resource endpoint URL from which embeddings should be retrieved. It should be in the format of https://YOUR_RESOURCE_NAME.openai.azure.com/openai/deployments/YOUR_DEPLOYMENT_NAME/embeddings. The api-version query parameter is not allowed.")
   endpoint: url;
 
+  // `authentication` according to the 05-01-preview spec, only supports api-key and data access token auth.
```

Comment:
It seems that specifically for vector search we can only use a subset of the possible authentication types. Reference: https://github.com/Azure/azure-rest-api-specs/pull/29042/files#diff-c48fe2c850e2a4161ac3a97642fc486eb977ab775fb8a2c9aa2b7b709213b35dR3353-R3375